src/store/flow/resource.ts

```diff
@@ -309,3 +310,40 @@ export function discoverContentTypesFor(mode: SupportedMode, resource?: IResourc
   }, contentTypeOverrides)
   return Object.assign(defaultModeMappings, contentTypeOverrides)[mode]
 }
+
+function createSetValidator<T>(allowedValues: T[]) {
+  return (lookup: T | T[]) => {
+    const lookupValues = castArray(lookup)
+
+    for (let i = 0; i < lookupValues.length; i += 1) {
+      if (allowedValues.includes(lookupValues[i])) {
+        return true
+      }
+    }
+    return false
+  }
+}
```

Comment:
This is a smart function 🙂

But I'm thinking, is it maybe too complex for the task? (It is complex because it 1) returns another function and 2) can handle both arrays and individual elements). Maybe it's not too hard per se, but it is a new pattern we haven't used. Also, I wonder if this function will ever be used again (is it worth it).

Do you think inlining the logic and using `flow.supported_modes.some(mode => value.modes.includes(mode))` or `_.intersection([],[]).length` as alternatives could be simpler?